lib/ansible/modules/cloud/azure/azure_rm_securitygroup.py

```diff
@@ -400,7 +400,7 @@ def compare_rules(r, rule):
         matched = True
         if rule.get('description', None) != r['description']:
             changed = True
-            r['description'] = rule['description']
+            r['description'] = rule.get('description', None)
         if rule['protocol'] != r['protocol']:
```

Comment:
will make the `protocol` also positional args?

Reply:
is there any other parameter need same fix so that user will be able to update security group eventually

Reply:
other params will be always there afaik